### ISSUE TITLE
Tree crown delineation

### ISSUE THREAD

**Issue by giswqs** [opengeos/segment-geospatial#92]
It would be great support this
https://github.com/PatBall1/detectree2

**Comment by Dseal95** [comment]
@giswqs could you please elaborate some more on this request? Would the goal be to have `detectree2` as a supported `model_type (str, optional): The model type.` in either `SamGeo` or  `SamGeo2` models? Thanks.

**Comment by Dseal95** [comment]
Hi @giswqs, just adding to my previous comment.

I used the `tree_mapping.ipynb` NB with `detectree2` to see where I could use it in the workflow. I have some examples to help with my previous comment. From my side, the most obvious use case for `detectree2` is to automate the tree vectors for SAM to them segment. I have attached examples from my NB below:

Imported vectors hardcoded from in the `tree_mapping.ipynb` with

```
geojson = (
    "https://github.com/opengeos/datasets/releases/download/samgeo/tree_boxes.geojson"
) 
```
<img width="529" alt="Image" src="https://github.com/user-attachments/assets/25ead7c7-f029-4461-9b98-aff5f7cff035" />

running `samgeo` on these vectors you get the following output:

<img width="515" alt="Image" src="https://github.com/user-attachments/assets/e8782bfb-756d-4916-92d8-49f5dcd39c14" />

`samgeo` performs well but there are obvious areas of FPs. 

Switching to `detectree2`, you can autogenerate the vectors. It is quite simple:

```
buffer = 20
tile_width = 45
tile_height = 45
tiles_path = "output_tiles"

tile_data(
    img_path=image,
    out_dir="output_tiles",
    buffer=buffer,
    tile_width=tile_width,
    tile_height=tile_height,
    dtype_bool=True,
)

# load the trained model.
if not os.path.exists("250312_flexi.pth"):
    !wget https://zenodo.org/records/15014353/files/250312_flexi.pth  # uncomment if not already downloaded.

trained_model = "./250312_flexi.pth"  # from: https://github.com/PatBall1/detectree2/tree/master/model_garden (An RGB model that is trained on both closed canopy systems and urban environments.)
cfg = setup_cfg(update_model=trained_model)

# CPU for NB purposes.
cfg.MODEL.DEVICE = "cpu"

# create .json predictions.
predict_on_data(directory=tiles_path, predictor=DefaultPredictor(cfg))

# convert .json predictions to .geojson
project_to_geojson(
    tiles_path=tiles_path,
    pred_fold=os.path.join(tiles_path, "predictions"),
    output_fold=os.path.join(tiles_path, "predictions", "predictions_geo"),
)

# run Detectree2 post-processing.
crowns = stitch_crowns(os.path.join(tiles_path, "predictions", "predictions_geo"), 1)

crowns = crowns[crowns.is_valid]

# simplify crowns to help with editing
simplify_thd = 0.3
crowns = crowns.set_geometry(crowns.simplify(simplify_thd))

# clean crowns.
iou_thd = 0.8
confidence_thd = 0.5  # lower this value to generate more vectors for the crowns.
crowns = clean_crowns(crowns=crowns, iou_threshold=iou_thd, confidence=confidence_thd)

# export to combined .geojson.
output_geojson = "dtrees.geojson"
crowns.to_file(filename=output_geojson, driver="GeoJSON")
```

The generated vectors are:

<img width="513" alt="Image" src="https://github.com/user-attachments/assets/639e5917-5c84-43ae-883d-4f3310890fbd" />

Which SAM then segments really well:

<img width="519" alt="Image" src="https://github.com/user-attachments/assets/e3bdef87-9009-4d9c-8bac-1c422b773931" />

There are some missed trees but it does avoid any manual labelling. You can lower the `detectree2` threshold `confidence_thd` to generate more vectors albeit with less confidence.

**Comment by giswqs** [comment]
Thank you for providing the great example. Did you generate bboxes from the `detectree2` output vector then pass them to SAMGeo?

**Comment by Dseal95** [comment]
@giswqs yeah exactly.  I passed the `image.tif` from the `tree_mapping.ipynb` directly into `detectree2` to generate a `geojson` of all the vectors. I then passed this `geojson` directly into SAM to generate masks. My intention was to demonstrate a rival workflow to the existing one which was; download `geojson` of manually-labelled vectors, pass to SAM for segmentation.
